### PR TITLE
Add dry run and diff

### PR DESCRIPTION
**Pull request by gene1wood**

### CODE REVIEW

File: uploader_rules.py [hunk 1]
```diff
@@ -102,7 +104,11 @@ def __init__(self, dct):
             remote_rule = remote_rules[rule_nr[0]]
             rule.is_the_same = (rule.script == remote_rule.get('script')) & (rule.enabled == bool(remote_rule.get('enabled'))) \
                    & (rule.stage == remote_rule.get('stage')) & (rule.order == remote_rule.get('order'))
-
+            if not rule.is_the_same:
+                logger.debug('Difference found in {} :'.format(rule.name))
+                for line in difflib.unified_diff(remote_rule.get('script').splitlines(), rule.script.splitlines(), fromfile='auth0-%s' % rule.name,
```
Comment (gdestuynder) [inline uploader_rules.py]: use `format`

File: uploader_rules.py [hunk 2]
```diff
@@ -131,18 +137,19 @@ def __init__(self, dct):
...
 
     ## Update & Create (I believe this may be atomic swaps for updates)
     for r in local_rules:
         if r.is_new:
             logger.debug("[+] Creating new rule {} on Auth0".format(r.name))
-            ret = authzero.create_rule(r)
-            logger.debug("+ New rule created with id {}".format(ret.get('id')))
+            if not args.dry_run:
+                ret =  authzero.create_rule(r)
+                logger.debug("+ New rule created with id {}".format(ret.get('id')))
```
Comment (gdestuynder) [inline uploader_rules.py]: should probably add the `dry_run` status in the log messages (and other log messages) for clarity

Comment (gene1wood) [reply uploader_rules.py]: This one is still pending. I'll see if I can put another commit in that adds this as it's a good idea. I'll hold on merging.

File: uploader_rules.py (hunk 1 above)
Comment (gene1wood) [reply uploader_rules.py]: Good idea. Fixed in 913b8dbccc93478af7aa235e23536446545f687b

**Comment by gdestuynder** [comment]
at this point we could almost just make it a function :)
anyway, lgtm feel free to merge

**Pull request merged**